Fast Pair: publish Model IDs and update the documentation (#11114)
nrfconnect/sdk-nrf: samples/bluetooth/peripheral_fast_pair/README.rst

Fast Pair device registration
=============================

Before you can use your device as a Fast Pair Provider, the device model must be registered with Google.
This is required to obtain Model ID and Anti-Spoofing Private Key.
You can register your own device or use the debug Model ID and Anti-Spoofing Public/Private Key pair obtained by Nordic for development purposes.
By default, if Model ID and Anti-Spoofing Private Key are not specified, the following debug Fast Pair provider is used:

* NCS Fast Pair demo - The input device Fast Pair provider:

   * Device name: NCS Fast Pair demo
   * Model ID: ``0x2A410B``
   * Anti-spoofing key (base64, uncompressed): ``Unoh+nycK/ZJ7k3dHsdcNpiP1SfOy0P/Lx5XixyYois=``
   * Type: Input device

>    * Additional features: Data only connection, Support personalized name, Ringing device unsupported

Reviewer: Can we use some kind of template doc mechanism for including this part of the documentation? We would need to have the possibility to provide some input arguments to it (e.g. Model ID = 0x2A410B, key, type, features)

The idea is to reuse the template in other places (e.g. nRF Desktop) and have a uniform way of phrasing this part of the documentation that is consistent across all projects.

@annwoj, @divipillai, is it possible?

Reviewer 2: yeah, I had the same idea. We can have include.txt for this content. If this cannot be fixed now, we can fix this in a separate PR.

Author: Good idea, I will work on it on monday, but if I don't manage to do it before release I will do it later.

.. include:: /includes/build_and_run_ns.txt

When building the sample, you can provide the Fast Pair Model ID (:c:macro:`FP_MODEL_ID`) and the Fast Pair Anti-Spoofing Key (:c:macro:`FP_ANTI_SPOOFING_KEY`) as CMake options.

> If the data is not provided, the default provisioning data, obtained for the "NCS Fast Pair demo", the input device debug Fast Pair provider, is used.

Reviewer 3: ```suggestion
If the data is not provided, the sample uses the default provisioning data obtained for the "NCS Fast Pair demo" (the input device debug Fast Pair provider).
```

@alstrzebonski Tired to rephrase this part a bit, please let me know in case it affects the meaning

Author: It's okay. Sounds better.

#. On the Android device, go to :guilabel:`Settings` > :guilabel:`Google` > :guilabel:`Devices & sharing` (or :guilabel:`Device connections`, depending on your Android device configuration) > :guilabel:`Devices`.
#. Move the Android device close to the Fast Pair Provider that is advertising.
#. Wait for Android device's notification about the detected Fast Pair Provider.
   If you use Model ID certified by Google, the notification is similar to the following one:

   .. figure:: /images/bt_fast_pair_discoverable_notification.png
      :scale: 33 %
      :alt: Fast Pair discoverable advertising Android notification

   The device model name and displayed logo depend on the data provided during the device model registration.

   If you use debug Model ID (e.g. default "NCS Fast Pair demo"), the notification is similar to the following one:

>    .. figure:: /images/bt_fast_pair_discoverable_notification_debug.png

Reviewer 3: @divipillai Should regular images have any specific frames?

Reviewer 2: yeah .png needs to be Width: 780 px and Height: 800 px as per the guidelines.

Author: Those are screenshots so the proportions (height to width ratio) depend really on what I want to show in the picture. Widening or lengthening the picture would look unnatural. Am I missing something?

Reviewer 3: In this case I think it's fine if we just make sure that the image parameters don't exceed the sizes mentioned by Divya. Would be great to get them closer to these parameters but without affecting the original shape of the image

   #. If the device does not appear on the list, wait until the data is synced between phones.

#. Move the second Android device close to the Fast Pair Provider.
   If the device is in the show UI indication advertising mode, a notification similar to the following appears:

   .. figure:: /images/bt_fast_pair_not_discoverable_notification.png
      :scale: 50 %
      :alt: Fast Pair not discoverable advertising Android notification

   or

   .. figure:: /images/bt_fast_pair_not_discoverable_notification_debug.png
      :scale: 40 %
      :alt: Fast Pair not discoverable advertising Android notification for debug Model ID

>    if you use debug Model ID.

Reviewer 3: ```suggestion
```
Should be merged with the beginning of the sentence above

>    The "Nordic" is replaced by your own Google account name as this is a default name written by the Fast Pair Seeker during initial pairing.

Reviewer 3: ```suggestion
   The "Nordic" name is replaced by your own Google account name as this is a default name created by the Fast Pair Seeker during the initial pairing.
```
